specification/cognitiveservices/data-plane/Language/preview/2022-05-01-preview/analyzeconversations.json

```diff
@@ -0,0 +1,2383 @@
+      "description": "The base class of a conversation input task.",
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/AnalyzeConversationTaskKind"
+        }
+      }
+    },
+    "AnalyzeConversationLROTask": {
+      "description": "The base class for an long running conversation input task.",
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "properties": {
+        "kind": {
+          "$ref": "#/definitions/AnalyzeConversationLROTaskKind"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "common.json#/definitions/TaskIdentifier"
+        }
+      ]
+    },
+    "AnalyzeConversationLROTaskKind": {
+      "type": "string",
+      "description": "Enumeration of supported analysis tasks on a collection of conversations.",
+      "enum": [
+        "ConversationalPIITask",
+        "ConversationalSummarizationTask"
+      ],
+      "x-ms-enum": {
+        "name": "AnalyzeConversationLROTaskKind",
+        "modelAsString": true
+      }
+    },
+    "AnalyzeConversationResultsKind": {
+      "type": "string",
+      "description": "Enumeration of supported Conversation Analysis task results.",
+      "enum": [
+        "ConversationalPIIResults",
+        "ConversationalSummarizationResults"
+      ],
+      "x-ms-enum": {
+        "name": "AnalyzeConversationResultsKind",
+        "modelAsString": true
+      }
+    },
+    "AnalyzeConversationPIITask": {
+      "type": "object",
+      "description": "Task definition for a PII redaction in conversations.",
+      "properties": {
+        "parameters": {
+          "$ref": "#/definitions/ConversationPIITaskParameters"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/AnalyzeConversationLROTask"
+        }
+      ],
+      "x-ms-discriminator-value": "ConversationalPIITask"
+    },
+    "ConversationPIIItemResult": {
+      "type": "object",
+      "required": [
+        "id",
+        "redactedContent",
+        "entities"
+      ],
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "redactedContent": {
+          "type": "object",
+          "$ref": "#/definitions/RedactedTranscriptContent"
+        },
+        "entities": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Entity"
+          }
+        }
+      }
+    },
+    "MultiLanguageConversationAnalysisInput": {
+      "required": [
+        "conversations"
+      ],
+      "properties": {
+        "conversations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Conversation"
+          }
+        }
+      }
+    },
+    "Conversation": {
+      "type": "object",
+      "description": "A complete ordered set of utterances (spoken or written), by one or more speakers to be used for analysis.",
+      "discriminator": "modality",
```

Comment:
Normally the term `kind` is preferred for defining a discriminator property according to the Azure API Guidelines:  https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#polymorphic-types

Reply:
@heaths what is your recommendation here. The reason to call the field "modality" is that it conveys the modality of the conversation to the customer and not just a discriminator.

Reply:
I've never been a fan of modality as it's not the word most people would naturally use for that concept.  Things like `inputMode`, `inputKind`, `inputFormat`, etc., resonate better with the average developer.

Reply:
Just a note on timing. I'd like to try and get all the feedback through and merge everything this week since we need to lock down the API for the May release. Any additional feedback is welcomed but will need to be implemented for the next preview iteration in July.

Reply:
Keeping it as "InputModality" to keep in sync with SDK.